Automattic/wp-calypso
# Posts: Add posts Redux state handlers

Related: #708, #2167
Prerequisite of: #303 

This pull request seeks to add new basic posts querying handlers to the global Redux state. This will be used in the post editor in enabling a user to search for existing content to link.

__Implementation notes:__

A few choices worth scrutinized review:

- I chose to index posts by `global_ID`, so that the `items` map could be a simple flattened object. Since post IDs are unique to a specific site, `global_ID` enables us to keep all known posts more readily accessible.
- "Request" as a chosen verb instead of "Fetch". There was some [previous discussion](https://github.com/Automattic/wp-calypso/pull/1660/files#r47776176) in which a concern was raised over using fetch in action names, as network requests aren't always fetch-related.
- Multiple queries can be tracked simultaneously, and are referenced by its JSON serialized query (where a query is the object passed to wpcom.js `postsList` method). Aside from allowing multiple post queries to be performed, this also prevents unnecessary re-requests when switching back to a previously known query result.

__Testing instructions:__

Confirm that Mocha tests pass by running `make run` in the project root directory, or directly from `client/state/posts`.

The included actions are not currently used anywhere in the application, though you should confirm that the inclusion of the `posts` reducer does not cause any errors to occur when loading any Calypso page.
